packages/core/src/browser/label-provider.ts

```diff
@@ -106,6 +143,20 @@ export namespace URIIconReference {
 @injectable()
 export class DefaultUriLabelProviderContribution implements LabelProviderContribution {
 
+    protected formatters: ResourceLabelFormatter[] = [];
+    protected readonly onDidChangeEmitter = new Emitter<DidChangeLabelEvent>();
+    protected homePath: string | undefined;
+    constructor(@inject(EnvVariablesServer) protected readonly envVariablesServer: EnvVariablesServer) {
+        const setHomePath = (variable: EnvVariable | undefined) => {
+            if (variable && variable.value) {
+                this.homePath = variable.value;
```

Comment:
good thinking, but we shoudl also fire an event here to ask clients to rerender with home path known

Reply:
done